README.md

```diff
@@ -36,32 +36,73 @@ The inspector runs both a client UI (default port 5173) and an MCP proxy server
 CLIENT_PORT=8080 SERVER_PORT=9000 npx @modelcontextprotocol/inspector node build/index.js
 ```
 
-For more details on ways to use the inspector, see the [Inspector section of the MCP docs site](https://modelcontextprotocol.io/docs/tools/inspector). For help with debugging, see the [Debugging guide](https://modelcontextprotocol.io/docs/tools/debugging).
+### Using a Configuration File
 
-### From this repository
-
-If you're working on the inspector itself:
-
-Development mode:
+The inspector supports configuration files to store settings for different MCP servers. This is useful when working with multiple servers or complex configurations:
 
 ```bash
-npm run dev
+npx @modelcontextprotocol/inspector --config path/to/config.json --server everything
+```
+
+Example configuration file:
+
+```json
+{
+  "mcpServers": {
+    "everything": {
+      "command": "npx",
+      "args": ["@modelcontextprotocol/server-everything"],
+      "env": {
+        "hello": "Hello MCP!"
+      }
+    },
+    "my-server": {
+      "command": "node",
+      "args": ["build/index.js", "arg1", "arg2"],
+      "env": {
+        "key": "value",
+        "key2": "value2"
+      }
+    }
+  }
+}
 ```
 
-> **Note for Windows users:**  
-> On Windows, use the following command instead:
->
-> ```bash
-> npm run dev:windows
-> ```
+### CLI Mode
 
-Production mode:
+CLI mode enables programmatic interaction with MCP servers from the command line, ideal for scripting, automation, and integration with coding assistants. This creates an efficient feedback loop for MCP server development.
 
 ```bash
-npm run build
-npm start
+npx @modelcontextprotocol/inspector --cli node build/index.js
 ```
 
-## License
+The CLI mode supports most operations across tools, resources, and prompts. A few examples:
+
+```bash
+# Basic usage
+npx @modelcontextprotocol/inspector --cli node build/index.js
+
+# With config file
+npx @modelcontextprotocol/inspector --cli --config path/to/config.json --server myserver
+
+# List available tools
+npx @modelcontextprotocol/inspector --cli node build/index.js --method tools/list
+
+# Call a specific tool
+npx @modelcontextprotocol/inspector --cli node build/index.js --method tools/call --tool-name mytool --tool-arg key=value --tool-arg another=value2
```

Comment:
instead of passing `tool-arg`(s) over cli, does it make sense to take a json file as input for passing arguments ? Considering that people will have to escape things properly on the command line, or hit into command length limitations, etc.